rama-http/src/service/web/router.rs

```diff
@@ -0,0 +1,276 @@
+use std::{convert::Infallible, sync::Arc};
+
+use crate::{
+    Request, Response,
+    matcher::{HttpMatcher, UriParams},
+};
+
+use matchit::Router as MatchitRouter;
+use rama_core::{
+    Context,
+    context::Extensions,
+    matcher::Matcher,
+    service::{BoxService, Service, service_fn},
+};
+use rama_http_types::{Body, IntoResponse, StatusCode};
+
+use super::IntoEndpointService;
+
+pub struct Router<State> {
+    routes: MatchitRouter<
+        Vec<(
+            HttpMatcher<State, Body>,
+            Arc<BoxService<State, Request, Response, Infallible>>,
+        )>,
+    >,
+    not_found: Arc<BoxService<State, Request, Response, Infallible>>,
+}
+
+impl<State> std::fmt::Debug for Router<State> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Router").finish()
+    }
+}
+
+impl<State> Router<State>
+where
+    State: Send + Sync + 'static,
+{
+    pub fn new() -> Self {
+        Self {
+            routes: MatchitRouter::new(),
+            not_found: Arc::new(
```

Comment:
I would just make this optional, and return a `404` if none.

but just for educational purposes thanks to <https://docs.rs/rama/latest/rama/http/service/web/trait.IntoEndpointService.html> you can just do `StatusCode::NOT_FOUND.into_endpoint_service()`... But again I would just Option it given you anyway box it already so easy enough to do.